daskernetes/objects.py

```diff
@@ -0,0 +1,167 @@
+                merge(a[key], b[key], path + [str(key)])
+            elif a[key] == b[key]:
+                pass # same leaf value
+            elif isinstance(a[key], list) and isinstance(b[key], list):
+                for idx, val in enumerate(b[key]):
+                    a[key][idx] = merge(a[key][idx], b[key][idx], path + [str(key), str(idx)], update=update)
+            elif update:
+                a[key] = b[key]
+            else:
+                raise Exception('Conflict at %s' % '.'.join(path + [str(key)]))
+        else:
+            a[key] = b[key]
+    return a
+
+def make_pod_spec(
+        image,
+        labels={},
+        threads_per_worker=1,
+        env={},
+        extra_container_config={},
+        extra_pod_config={},
+        memory_limit=None,
+        memory_request=None,
+        cpu_limit=None,
+        cpu_request=None,
+):
+    """
+    Create a pod template from various parameters passed in.
+    """
+    pod = client.V1Pod(
+        metadata=client.V1ObjectMeta(
+            labels=labels
+        ),
+        spec=client.V1PodSpec(
+            restart_policy='Never',
+            containers=[
+                client.V1Container(
+                    name='dask-worker',
+                    image=image,
+                    args=[
+                        'dask-worker',
+                        '$(DASK_SCHEDULER_ADDRESS)',
+                        '--nthreads', str(threads_per_worker),
+                    ],
+                    env=[client.V1EnvVar(name=k, value=v)
+                            for k, v in env.items()],
+                )
+            ]
+        )
+    )
+
+
+    resources = client.V1ResourceRequirements(limits={}, requests={})
+
+    if cpu_request:
+        resources.requests['cpu'] = cpu_request
+    if memory_request:
+        resources.requests['memory'] = memory_request
+
+    if cpu_limit:
+        resources.limits['cpu'] = cpu_limit
+    if memory_limit:
+        resources.limits['memory'] = memory_limit
+
+    pod.spec.containers[0].resources = resources
+
+    for key, value in extra_container_config.items():
+        _set_k8s_attribute(
+            pod.spec.containers[0],
+            key,
+            value
+        )
+
+    for key, value in extra_pod_config.items():
+        _set_k8s_attribute(
+            pod.spec,
+            key,
+            value
+        )
+    return pod
+
+
+class _FakeResponse:
+    def __init__(self, data):
+        self.data = json.dumps(data)
+def make_pod_from_yaml(filepath):
+    if not yaml:
+        raise ImportError("PyYaml is required to use yaml functionality, please install!")
+
+    apiclient = client.ApiClient()
+    with open(filepath) as f:
+        # FIXME: We can't use the 'deserialize' function since
+        # that expects a response object!
+        pod_template = apiclient.deserialize(
+            _FakeResponse(yaml.safe_load(f)),
+            client.V1Pod
+        )
```

Comment:
I wasn't able to make this approach work before.  A number of components of the kubernetes object were silently dropped.  I had a bunch of hacks on top of this to get my use case to work.  Those hacks were pretty fragile.  https://github.com/mrocklin/daskernetes/blob/deploy/daskernetes/core.py#L338-L374

Reply:
This particular problem was my main blocker.  I wasn't familiar enough with the `kubernetes` library to easily find a path out.  Most other issues I'm able to handle